app/controllers/api_controller/provision_requests.rb

```diff
@@ -18,5 +18,25 @@ def create_resource_provision_requests(type, _id, data)
       MiqProvisionVirtWorkflow.from_ws(version_str, @auth_user_obj, template_fields, vm_fields, requester, tags,
                                        additional_values, ems_custom_attrs, miq_custom_attrs)
     end
+
+    def deny_resource_provision_requests(type, id, data)
+      provreq = resource_search(id, type, collection_class(:provision_requests))
+      provreq.deny(@auth_user, data['reason'])
+      res = action_result(true, "Provision request #{id} denied")
+      add_href_to_result(res, type, id)
```

Comment:
It could be nice to have the href, I will try to have a cleaner way thanks to your input asap.

Thanks you !